oap-server/server-alarm-plugin/src/main/java/org/apache/skywalking/oap/server/core/alarm/provider/EventHookCallback.java

```diff
@@ -0,0 +1,158 @@
+import org.apache.skywalking.oap.server.core.analysis.IDManager;
+import org.apache.skywalking.oap.server.core.source.DefaultScopeDefine;
+import org.apache.skywalking.oap.server.library.module.ModuleManager;
+
+import java.util.List;
+import java.util.Objects;
+import java.util.UUID;
+
+/**
+ * EventCallBack: When an alert is present, an event is generated for each alert message. These events are then sent to the internal event analyzer.
+ *
+ */
+public class EventHookCallback implements AlarmCallback {
+
+    private final ModuleManager manager;
+
+    public EventHookCallback(ModuleManager manager) {
+        this.manager = manager;
+    }
+
+    @Override
+    public void doAlarm(List<AlarmMessage> alarmMessage) {
+        if (Objects.isNull(this.manager)) {
+            return ;
+        }
+        EventAnalyzerService analyzerService = manager.find(EventAnalyzerModule.NAME).provider().getService(EventAnalyzerService.class);
+        alarmMessage.forEach(a -> {
+            for (Event event : constructCurrentEvent(a)) {
+                if (Objects.nonNull(event)) {
+                    analyzerService.analyze(event);
+                }
+            }
+        });
+    }
+
+    private Event[] constructCurrentEvent(AlarmMessage msg) {
+        Event[] events = new Event[2];
+        long now = System.currentTimeMillis();
+        Event.Builder builder = Event.newBuilder()
+                .setUuid(UUID.randomUUID().toString())
+                .setName("Alarm")
+                .setStartTime(now - (msg.getPeriod() * 60 * 1000))
+                .setMessage(msg.getAlarmMessage())
+                .setType(Type.Error)
+                .setEndTime(now);
+        switch (msg.getScopeId()) {
+            case DefaultScopeDefine.SERVICE :
+                IDManager.ServiceID.ServiceIDDefinition singleServiceIdDef = IDManager.ServiceID.analysisId(msg.getId0());
+                builder.setSource(
+                    Source.newBuilder()
+                            .setService(singleServiceIdDef.getName())
+                            .build()
+                );
+                events[0] = builder.build();
+                break;
+            case DefaultScopeDefine.SERVICE_RELATION :
+                IDManager.ServiceID.ServiceIDDefinition doubleServiceIdDef = IDManager.ServiceID.analysisId(msg.getId0());
+                builder.setSource(
+                        Source.newBuilder()
+                            .setService(doubleServiceIdDef.getName())
+                            .build()
+                );
+                events[0] = builder.build();
+                doubleServiceIdDef = IDManager.ServiceID.analysisId(msg.getId1());
+                builder.setSource(
+                        Source.newBuilder()
+                                .setService(doubleServiceIdDef.getName())
+                                .build()
+                ).setUuid(UUID.randomUUID().toString());
```

Comment:
Why set UUID again?

Reply:
> Why set UUID again?

https://github.com/apache/skywalking/pull/6702#discussion_r611450353 when building the second event, the UUID need to be regenerated otherwise it would override the first event (or is dropped)

Reply:
Oh, I see.